src/bokeh/palettes.py

```diff
@@ -1639,6 +1639,43 @@ def set_alpha(rgba: RGB, i: int) -> RGB:
 
     return palette
 
+def interp_palette(palette: Palette, n: int) -> Palette:
+    """ Generate a new palette by interpolating a given palette.
+
+    Linear interpolation is performed separately on each of the RGBA
+    components.
+
+    Args:
+        palette (seq[str]) :
+            A sequence of hex RGB(A) color strings to create new palette from
+
+        n (int) :
+            The size of the palette to generate
+
+    Returns:
+        seq[str] : a sequence of hex RGB(A) color strings
+
+    Raises:
+        ValueError if n is negative
+
+    """
+    npalette = len(palette)
+    if npalette < 1:
+        raise ValueError("palette must contain at least one color")
+    if n < 0:
+        raise ValueError("requested palette length cannot be negative")
+
+    rgba_array = to_rgba_array(palette)
+    integers = np.arange(npalette)
+    fractions = np.linspace(0, npalette-1, n)
+
+    r = np.interp(fractions, integers, rgba_array[:, 0]).astype(np.uint8)
+    g = np.interp(fractions, integers, rgba_array[:, 1]).astype(np.uint8)
+    b = np.interp(fractions, integers, rgba_array[:, 2]).astype(np.uint8)
+    a = np.interp(fractions, integers, rgba_array[:, 3]) / 255  # Remains floating-point
```

Comment:
I would expect numpy to follow Python's semantics for division operator. I would also probably use `255.0` constant as an indicator that this is a floating-point computation. Not a rule yet, but I'm getting to like this in Rust very much.

Reply:
`255.0` it is then.